backend/test/modules/email-sending/core/application/SendEmail.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { testSendEmailModule } from './TestSendEmailModule';
+import { SendEmail } from '../../../../../src/modules/email-sending/core/application/command/SendEmail';
+
+describe('Send an email to player', () => {
+  it('When command SendEmail is sent, then send an email with appropriate arguments', async () => {
+    //Given
+    const sendEmailModuleCore = testSendEmailModule();
+    const sendEmail = new SendEmail({
+      subject: 'test',
+      emailAddress: 'test@gmail.com',
+      htmlContent: '<div>test</div>',
+    });
+
+    //When
+    const commandResult = await sendEmailModuleCore.executeCommand(sendEmail);
+
+    //Then
+    expect(commandResult.isSuccess()).toBeTruthy();
```

Comment:
i w tym teście też nie mam pojęcia jak przetestować, że została wywołana funkcja wysyłania maili? Wgl która funkcja i z czego dokładnie? 
Zapewne chodzi o funkcję sendAnEmail z interfejsu EmailSender, ale jak wgl się do niej odnieść?

Reply:
Jeśli to jest Unit Test, a nie integracyjny (np. z mailhogiem / bazą danych) to nie możesz używać NodeMailerEmailSender, bo bez dockera działającego Ci nie przejdzie. Możesz przekazac emailSender jako argument do testSendEmailModule i w unit teście mockować.

Reply:
https://github.com/nowakprojects/CodersCamp2020.Project.FullStack-Node-React.TableSoccerTournaments/pull/37/commits/acf8b58f1fc6f1fa043bdfa518ea915b74e3ef87
teraz chyba jest to o co chodziło :-) ?